app/scripts/modules/core/src/presentation/modal/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+
+const { useMemo } = React;
+
+import { useEventListener, useContainerClassNames, useLatestCallback } from 'core/presentation';
+
+import { ModalContext } from './ModalContext';
+import * as styles from './Modal.module.css';
+
+export interface IModalProps {
+  isOpen: boolean;
+  onRequestClose?: () => any;
+  children?: React.ReactNode;
+}
+
+export const Modal = ({ onRequestClose, isOpen, children }: IModalProps) => {
+  useContainerClassNames(isOpen ? [styles.backdropBlurEffect] : []);
+
+  const keydownCallback = ({ keyCode }: KeyboardEvent) => {
+    if (keyCode === 27) {
+      onRequestClose();
+    }
+  };
+  useEventListener(document, 'keydown', isOpen ? keydownCallback : null);
+
+  const memoizedRequestClose = useLatestCallback(onRequestClose);
+  const modalContext = useMemo(() => ({ onRequestClose: memoizedRequestClose }), []);
```

Comment:
I believe this `useMemo` is unnecessary?

Reply:
Sadly it is necessary, because context providers re-render whenever the _reference_ of their `value` prop changes ([docs](https://reactjs.org/docs/context.html#caveats)). If you pass in an object literal that gets generated inside render, you end up with a chain reaction where every time the parent component renders you get a second render down the whole subtree underneath the context provider 😢